app/models/concerns/spree/gift_cards/order_contents_concerns.rb

```diff
@@ -0,0 +1,71 @@
+module Spree
+  module GiftCards::OrderContentsConcerns
+    extend ActiveSupport::Concern
+
+    included do
+      prepend(InstanceMethods)
+    end
+
+    module InstanceMethods
+      def add(variant, quantity = 1, options = {})
+        line_item = super
```

Comment:
don't take this as real feedback, just a thought if you wanted this is also an option:

```ruby
super.tap do |line_item|
  create_gift_cards(line_item, quantity, options["gift_card_details"] || {})
end
```